### ISSUE TITLE
Getting and Cleaning Data Module 2 Error (Grouping and Chaining with dplyr)

### ISSUE THREAD

**Issue by aluxh** [swirldev/swirl#276]
Course: Getting and Cleaning Data
Module 2: Grouping and Chaining with dplyr

Hi, I get the following error when I try to start module 2:

```
| Attemping to load lesson dependencies...

| Package ‘dplyr’ loaded correctly!

Error in rbind(deparse.level, ...) : 
  numbers of columns of arguments do not match
```

![screen shot 2015-03-13 at 9 42 19 pm](https://cloud.githubusercontent.com/assets/1736667/6639363/eb316c02-c9c9-11e4-992f-7d53d8379ce3.png)

OSX 10.10.2

sessionInfo:

```
R version 3.1.0 (2014-04-10)
Platform: x86_64-apple-darwin13.1.0 (64-bit)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tidyr_0.2.0 dplyr_0.4.1 swirl_2.2.9

loaded via a namespace (and not attached):
 [1] assertthat_0.1  DBI_0.3.1       digest_0.6.4    httr_0.3        lazyeval_0.1.10 magrittr_1.5    parallel_3.1.0  Rcpp_0.11.2    
 [9] RCurl_1.95-4.1  stringr_0.6.2   testthat_0.8.1  tools_3.1.0     yaml_2.1.13    
```

**Comment by WilCrofter** [comment]
@aluxh, this post will not be helpful to you. I'm posting mainly for the other developers.

The error occurred immediately after lesson selection, so loading dependencies and executing the initLesson.R file are suspect. Swirl reported that `dplyr` was loaded successfully, and it appears in sessionInfo. This implicates initLesson.R, but initialization worked for me so coding bugs are unlikely.

The data file is very large. Could it be problem involving heap size or something?

**Comment by aluxh** [comment]
Thanks @WilCrofter: Are there anything I can do to test on that? Strangely, it has the same when I go to module 3 too.

**Comment by WilCrofter** [comment]
@aluxh, I don't see any large files in [the third module](https://github.com/swirldev/swirl_courses/tree/master/Getting_and_Cleaning_Data/Tidying_Data_with_tidyr) so memory limitations are not the problem. Did you get that same error message, `Error in rbind(deparse.level, ...) :   numbers of columns of arguments do not match`, in module 3?

Swirl does use `rbind` during initialization. But, since I don't get the error on linux, and since the course was developed on OSX,  swirl's use of rbind seems unlikely to be the source of the error. The initLesson.R files for [both](https://github.com/swirldev/swirl_courses/blob/master/Getting_and_Cleaning_Data/Grouping_and_Chaining_with_dplyr/initLesson.R) [lessons](https://github.com/swirldev/swirl_courses/blob/master/Getting_and_Cleaning_Data/Tidying_Data_with_tidyr/initLesson.R) use `tbl_df` from package `dplyr`. You and I have the same version of `dplyr` but I notice that some of your packages are earlier versions than mine. Possibly, `dplyr` needs more current versions of one or more of these packages. Updating your packages would be worth a try.

My package versions are listed below. 

```
other attached packages:
[1] dplyr_0.4.1  swirl_2.2.21

loaded via a namespace (and not attached):
 [1] assertthat_0.1  bitops_1.0-6    DBI_0.3.1       digest_0.6.8    httr_0.6.1     
 [6] lazyeval_0.1.10 magrittr_1.5    parallel_3.1.2  Rcpp_0.11.4     RCurl_1.95-4.5 
[11] stringr_0.6.2   testthat_0.9.1  tools_3.1.2     yaml_2.1.13  
```

**Comment by ncarchedi** [comment]
You need to update swirl to 2.2.21. That will solve the problem.

**Comment by aluxh** [comment]
Thanks @ncarchedi! It works now!

**Comment by rajeshblue** [comment]
Hi,

When I chose the above lesson as an option, it shows the following options:

| Please choose a lesson, or type 0 to return to course menu.

1: 2014-07-08.csv

2: customTests.R

3: dependson.txt

4: initLesson.R

5: lesson.yaml

6: outline.R

7: scripts

None of the above options make any sense to me, so if I choose, lets say, 1, I get the following error:

------------------------

Error in file(con, "r") : invalid 'description' argument

| Leaving swirl now. Type swirl() to resume.

--------------------

In other words, I am unable to intiate the lesson. Anyone faced this?

Cheers

RK

**Comment by seankross** [comment]
Hi @rajeshblue,

It looks like you need to reinstall the course. Try the following:

```r
library(swirl)
uninstall_all_courses()
install_from_swirl("Getting and Cleaning Data")
swirl()
```

**Comment by rajeshblue** [comment]
Thanks @seankross. The oldest trick in the book worked!
